### PR TITLE
27 get/meals

### PR DESCRIPTION
**Pull request by SajedaIsmail**

### CODE REVIEW

File: server/controllers/getMeals.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+const getMealsByCat = require('../database/queries/getMealsByCat');
+
+const getMeals = (req, res, next) => {
+  let { category } = req.query;
```
Comment (ahmedisam99) [inline server/controllers/getMeals.js]: Add a default value for the category incase the URL params are not supplied
```suggestion
  let { category = 'main' } = req.query;
```

File: server/controllers/getMeals.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  if (category) category = category.toLowerCase();
+  getMealsByCat(category)
+    .then((result) => (result.rows.length
+        ? res.status(200).send({ code: 200, data: result.rows })
+        : res.status(400).send({
+            code: 400,
+            error: 'Bad Request: there is no such category'
```
Comment (ahmedisam99) [inline server/controllers/getMeals.js]:
```suggestion
            error: 'No such category'
```

File: server/controllers/getMeals.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+const getMealsByCat = require('../database/queries/getMealsByCat');
+
+const getMeals = (req, res, next) => {
+  let { category } = req.query;
+  if (category) category = category.toLowerCase();
+  getMealsByCat(category)
+    .then((result) => (result.rows.length
+        ? res.status(200).send({ code: 200, data: result.rows })
```
Comment (ahmedisam99) [inline server/controllers/getMeals.js]:
```suggestion
        ? res.status(200).send({ statusCode: 200, data: result.rows })
```

File: server/controllers/getMeals.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+        : res.status(400).send({
+            code: 400,
```
Comment (ahmedisam99) [inline server/controllers/getMeals.js]:
```suggestion
            statusCode: 400,
```

File: test/index.js [hunk 5]
```diff
@@ -1 +1,8 @@
+const tape = require('tape');
```
Comment (ahmedisam99) [inline test/index.js]: remove the 1 == 1 test

File: server/controllers/getMeals.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+const getMealsByCat = require('../database/queries/getMealsByCat');
+
+const getMeals = (req, res, next) => {
+  let { category = 'main' } = req.query;
+  if (category) category = category.toLowerCase();
+  getMealsByCat(category)
+    .then((result) => (result.rows.length
+      ? res.status(200).send({ statusCode: 200, data: result.rows })
```
Comment (ranasobeid95) [inline server/controllers/getMeals.js]: the status will be 200 by default

File: test/getMeals.test.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
+const supertest = require('supertest');
+const test = require('tape');
+const app = require('../server/app');
+
+module.exports = test('Testing for getMeals route', (t) => {
```
Comment (ahmedisam99) [inline test/getMeals.test.js]: remove the module.exports
```suggestion
test('Testing for getMeals route', (t) => {
```

File: test/getMeals.test.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+        200,
+        'The code value must be 200 in the response body',
+      );
+      res.body.data.forEach((ele) => t.equal(ele.category_id, 1, 'All category ids must be 1'));
+      res.body.data.forEach((ele) => t.equal(ele.category, 'main', 'All category names must be main'));
+      t.end();
+    });
+});
+
+module.exports = test('Testing for getMeals route', (t) => {
```
Comment (ahmedisam99) [inline test/getMeals.test.js]: remove module.exports
```suggestion
test('Testing for getMeals route', (t) => {
```

File: test/getMeals.test.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+      t.deepEqual(
+        res.body,
+        { statusCode: 400, error: 'No such category' },
+        'The code value must be 200 in the response body',
+      );
+      t.end();
+    });
+});
+
+test.onFinish(() => process.exit(0));
```
Comment (ahmedisam99) [inline test/getMeals.test.js]: remove this line
```suggestion

```

**Pull request merged**